integrations/nlu-microsoftclu/src/MicrosoftCluNlu.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  EntityMap,
+  InterpretationPluginConfig,
+  Jovo,
+  JovoError,
+  NluData,
+  NluPlugin,
+} from '@jovotech/framework';
+
+import {
+  AnalyzeConversationResult,
+  ConversationAnalysisClient,
+  ConversationAnalysisClientOptionalParams,
+  ConversationPrediction,
+  ConversationTaskParameters,
+  ConversationalTask,
+} from '@azure/ai-language-conversations';
+
+import { AzureKeyCredential, KeyCredential, TokenCredential } from '@azure/core-auth';
+import { v4 as uuidV4 } from 'uuid';
+import { INTENT_NONE, PROJECTKIND_CONVERSATION } from './constants';
+
+export interface MicrosoftCluLibraryConfig {
+  id: string;
+  participantId: string;
+  taskParameters: ConversationTaskParameters;
+  options?: ConversationAnalysisClientOptionalParams;
+}
+
+export interface MicrosoftCluNluConfig extends InterpretationPluginConfig {
+  endpoint: string;
+  credential: TokenCredential | KeyCredential | string;
+  fallbackLanguage: string;
+  libraryConfig: MicrosoftCluLibraryConfig;
+}
+
+export class MicrosoftCluNlu extends NluPlugin<MicrosoftCluNluConfig> {
+  getDefaultConfig(): MicrosoftCluNluConfig {
+    return {
+      ...super.getDefaultConfig(),
+      fallbackLanguage: 'en',
+      endpoint: '',
+      libraryConfig: {
+        id: uuidV4(),
```

Comment:
Isn't `id` and `participantId` more like request-related data? The current solution would result in the same ids for every request. 

I can't find more information about that in the CLU docs...

Reply:
I couldn't find more info about them either. Here is one of the few samples and they just set the value to "1":
https://learn.microsoft.com/en-us/rest/api/language/2022-05-01/conversation-analysis-runtime/analyze-conversation?tabs=HTTP#conversation-project-result

That's why I surfaced it in config but defaulted to a UUID

Reply:
Do you have a better idea of how to approach this?

Reply:
I posted a question on the Microsoft forum here:
https://learn.microsoft.com/en-us/answers/questions/1276259/when-making-an-api-request-to-clu-analysis-what-ar

Reply:
@aswetlow Looking at the most recent responses from the Microsoft forum, I think I'm going to remove the config settings for `id` and `participantId` and instead set the `id` value to `request.id` and the `participantId` to the string `"User"`.

What do you think?